src/webgpu/api/validation/capability_checks/features/texture_formats.spec.ts

```diff
@@ -171,6 +171,53 @@ g.test('canvas_configuration')
     }
   });
 
+g.test('canvas_configuration_view_formats')
+  .desc(
+    `
+  Test that configuring a canvas with view formats throws an exception if the required optional
+  feature is not enabled. Otherwise, a validation error should be generated instead of throwing an
+  exception.
+  `
+  )
+  .params(u =>
+    u
+      .combine('format', kOptionalTextureFormats)
+      .combine('canvasType', kAllCanvasTypes)
+      .combine('enable_required_feature', [true, false])
+  )
+  .beforeAllSubcases(t => {
+    const { format, enable_required_feature } = t.params;
+
+    const formatInfo = kTextureFormatInfo[format];
+    if (enable_required_feature) {
+      t.selectDeviceOrSkipTestCase(formatInfo.feature);
+    }
+  })
+  .fn(async t => {
+    const { format, canvasType, enable_required_feature } = t.params;
+
+    const canvas = createCanvas(t, canvasType, 2, 2);
+    const ctx = canvas.getContext('webgpu' as const);
+    assert(ctx !== null, 'Failed to get WebGPU context from canvas');
+
+    const canvasConf = {
+      device: t.device,
+      format,
```

Comment:
The test will pass even if you comment out the `viewFormats: [format]` line because the invalid format is also used here. This should be always set to some valid format since we only want to test `viewFormats`.
```suggestion
      format: 'bgra8unorm',
```

Reply:
Done.